src/main/java/org/checkerframework/specimin/TargetMemberFinderVisitor.java

```diff
@@ -246,6 +246,16 @@ public Visitable visit(ConstructorDeclaration method, Void p) {
           resolvedMethod.getPackageName() + "." + resolvedMethod.getClassName(),
           usedTypeElements,
           nonPrimaryClassesToPrimaryClass);
+      if (modularityModel.preserveAllFieldsIfTargetIsConstructor()) {
+        // This cast is safe, because a constructor must be contained in a class declaration.
+        ClassOrInterfaceDeclaration thisClass =
+            (ClassOrInterfaceDeclaration) JavaParserUtil.getEnclosingClassLike(method);
+        for (FieldDeclaration field : thisClass.getFields()) {
+          for (VariableDeclarator variable : field.getVariables()) {
+            usedMembers.add(currentClassQualifiedName + "#" + variable.getNameAsString());
```

Comment:
Do we need to make sure the field types are included in `usedTypeElements`? Currently `visit(VariableDeclarator)` only checks if it's in the `targetFields` set, not `usedMembers` so these types may not be included in the output

Reply:
Good catch! It's actually trickier than that - I also needed to trigger synthetic class generation for the types of the fields in `UnsolvedSymbolVisitor`. I've added a test for this case.